File: apps/mobile/lib/screens/note_screen.dart
import 'dart:convert';
import 'dart:io';
import 'package:dio/dio.dart';
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:image_picker/image_picker.dart';
import 'package:intl/intl.dart';
import 'package:http_parser/http_parser.dart';
import 'package:shared_preferences/shared_preferences.dart';

import '../config/app_config.dart';
import '../models/diary_request_model.dart';
import '../models/diary_response_model.dart';
import '../models/user_plant_model.dart';
import '../services/diary_api_service.dart';

class NoteScreen extends StatefulWidget {
  final DateTime selectedDate;
  final DiaryResponse? editingDiary; // 수정 모드 시 전달

  const NoteScreen({super.key, required this.selectedDate, this.editingDiary});

  @override
  State<NoteScreen> createState() => _NoteScreenState();
}

class _NoteScreenState extends State<NoteScreen> {
  final TextEditingController _titleController = TextEditingController();
  final TextEditingController _notesController = TextEditingController();
  final ImagePicker _picker = ImagePicker();

  List<UserPlantResponse> _myRegisteredPlants = [];
  String? _selectedPlantNickname;
  bool _isLoadingPlants = true;
  bool _isSaving = false;
  String? _error;

  XFile? _imageFile;
  bool _watered = false;
  bool _fertilized = false;
  bool _pruned = false;
  bool _deleteExistingImage = false;

  @override
  void initState() {
    super.initState();
    _fetchMyPlants().then((_) => _initEditModeIfNeeded());
  }

  void _initEditModeIfNeeded() {
    if (widget.editingDiary != null) {
      final d = widget.editingDiary!;
      _titleController.text = d.title;
      _notesController.text = d.content ?? '';
      _watered = d.watered;
      _fertilized = d.fertilized;
      _pruned = d.pruned;

      if (d.connectedUserPlantIds.isNotEmpty &&
          _myRegisteredPlants.isNotEmpty) {
        final userPlantId = d.connectedUserPlantIds.first;
        final plant = _myRegisteredPlants.firstWhere(
          (p) => p.userPlantId == userPlantId,
          orElse: () => _myRegisteredPlants.first,
        );
        _selectedPlantNickname = plant.plantNickname;
      } else {
        _selectedPlantNickname = null;
      }
    }
  }

  @override
  void dispose() {
    _titleController.dispose();
    _notesController.dispose();
    super.dispose();
  }

  Future<void> _fetchMyPlants() async {
    try {
      final SharedPreferences prefs = await SharedPreferences.getInstance();
      final accessToken = prefs.getString('accessToken');
      final dio = Dio(
        BaseOptions(
          baseUrl: AppConfig.apiBaseUrl,
          headers: {HttpHeaders.authorizationHeader: 'Bearer $accessToken'},
        ),
      );
      final response = await dio.get('/api/user-plants');
      final List<UserPlantResponse> plants = (response.data as List)
          .map((e) => UserPlantResponse.fromJson(e))
          .toList();
      if (mounted) setState(() => _myRegisteredPlants = plants);
    } on DioException catch (e) {
      if (mounted) setState(() => _error = "식물 목록 로딩 실패: ${e.message}");
    } finally {
      if (mounted) setState(() => _isLoadingPlants = false);
    }
  }

  Future<void> _pickImage() async {
    final pickedFile = await _picker.pickImage(
      source: ImageSource.gallery,
      imageQuality: 80,
    );
    if (pickedFile != null) setState(() => _imageFile = pickedFile);
  }

  Future<void> _saveNote() async {
    if (_isSaving) return;

    if (_selectedPlantNickname == null) {
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(
          content: Text('어떤 식물에 대한 기록인지 선택해주세요!', style: GoogleFonts.gaegu()),
        ),
      );
      return;
    }
    if (_titleController.text.trim().isEmpty) {
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(content: Text('제목은 꼭 입력해주세요!', style: GoogleFonts.gaegu())),
      );
      return;
    }

    setState(() => _isSaving = true);

    try {
      final selectedPlant = _myRegisteredPlants.firstWhere(
        (plant) => plant.plantNickname == _selectedPlantNickname,
      );

      if (selectedPlant.userPlantId == null) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(
            content: Text(
              '식물 정보에 문제가 있습니다. 관리자에게 문의하세요.',
              style: GoogleFonts.gaegu(),
            ),
          ),
        );
        setState(() => _isSaving = false);
        return;
      }

      final diaryRequest = DiaryRequest(
        selectedUserPlantIds: [selectedPlant.userPlantId!],
        title: _titleController.text.trim(),
        content: _notesController.text.trim(),
        diaryDate: DateFormat('yyyy-MM-dd').format(widget.selectedDate),
        watered: _watered,
        fertilized: _fertilized,
        pruned: _pruned,
        deleteExistingImage: _deleteExistingImage,
      );
      final String requestJson = jsonEncode(diaryRequest.toJson());

      final prefs = await SharedPreferences.getInstance();
      final accessToken = prefs.getString('accessToken');
      final dio = Dio(
        BaseOptions(
          baseUrl: AppConfig.apiBaseUrl,
          headers: {HttpHeaders.authorizationHeader: 'Bearer $accessToken'},
        ),
      );

      if (widget.editingDiary == null) {
        // 신규 등록
        final formData = FormData();
        formData.files.add(
          MapEntry(
            'diaryRequest',
            MultipartFile.fromString(
              requestJson,
              contentType: MediaType('application', 'json'),
              filename: 'data.json',
            ),
          ),
        );
        if (_imageFile != null) {
          final file = File(_imageFile!.path);
          formData.files.add(
            MapEntry(
              'imageFile',
              await MultipartFile.fromFile(
                file.path,
                filename: 'note.jpg',
                contentType: MediaType('image', 'jpeg'),
              ),
            ),
          );
        }
        final response = await dio.post('/api/diaries', data: formData);
        if (response.statusCode == 200 || response.statusCode == 201) {
          if (mounted) {
            ScaffoldMessenger.of(context).showSnackBar(
              SnackBar(
                content: Text('일지가 성공적으로 기록되었어요!', style: GoogleFonts.gaegu()),
                backgroundColor: Colors.green,
              ),
            );
            Navigator.pop(context, true);
          }
        } else {
          throw Exception("기록 실패: ${response.statusCode}");
        }
      } else {
        // 수정 모드 (PUT)
        final diaryId = widget.editingDiary!.diaryId;
        final formData = FormData();
        formData.files.add(
          MapEntry(
            'request',
            MultipartFile.fromString(
              requestJson,
              contentType: MediaType('application', 'json'),
              filename: 'data.json',
            ),
          ),
        );
        if (_imageFile != null) {
          final file = File(_imageFile!.path);
          formData.files.add(
            MapEntry(
              'newImageFile',
              await MultipartFile.fromFile(
                file.path,
                filename: 'note.jpg',
                contentType: MediaType('image', 'jpeg'),
              ),
            ),
          );
        }
        final response = await dio.put('/api/diaries/$diaryId', data: formData);
        if (response.statusCode == 200) {
          if (mounted) {
            ScaffoldMessenger.of(context).showSnackBar(
              SnackBar(
                content: Text('일지가 성공적으로 수정되었어요!', style: GoogleFonts.gaegu()),
                backgroundColor: Colors.green,
              ),
            );
            Navigator.pop(context, true);
          }
        } else {
          throw Exception("수정 실패: ${response.statusCode}");
        }
      }
    } on DioException catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(
            content: Text(
              '저장 실패: ${e.response?.data['message'] ?? e.message}',
              style: GoogleFonts.gaegu(),
            ),
            backgroundColor: Colors.redAccent,
          ),
        );
      }
    } finally {
      if (mounted) setState(() => _isSaving = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    String formattedDate = DateFormat(
      'yyyy. MM. dd (E)',
      'ko_KR',
    ).format(widget.selectedDate);

    return Scaffold(
      backgroundColor: const Color(0xFFFDFCF8),
      appBar: AppBar(
        title: Text(
          widget.editingDiary == null ? '오늘의 일지' : '일지 수정',
          style: GoogleFonts.gaegu(fontWeight: FontWeight.bold),
        ),
        backgroundColor: const Color(0xFF81C784),
        foregroundColor: Colors.white,
        elevation: 0,
        actions: [
          TextButton(
            onPressed: _isSaving ? null : _saveNote,
            child: _isSaving
                ? const SizedBox(
                    width: 20,
                    height: 20,
                    child: CircularProgressIndicator(
                      color: Colors.white,
                      strokeWidth: 2.5,
                    ),
                  )
                : Text(
                    widget.editingDiary == null ? '기록하기' : '수정하기',
                    style: GoogleFonts.gaegu(
                      color: Colors.white,
                      fontSize: 18,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
          ),
        ],
      ),
      body: _isLoadingPlants
          ? const Center(child: CircularProgressIndicator())
          : _error != null
          ? Center(
              child: Padding(
                padding: const EdgeInsets.all(16.0),
                child: Text(_error!, textAlign: TextAlign.center),
              ),
            )
          : SingleChildScrollView(
              padding: const EdgeInsets.all(20.0),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  _buildDateChip(formattedDate),
                  const SizedBox(height: 24),
                  _buildTitleField(),
                  const SizedBox(height: 24),
                  _buildMyPlantSelection(),
                  const SizedBox(height: 24),
                  _buildImagePicker(),
                  if (widget.editingDiary != null &&
                      widget.editingDiary!.imageUrl != null)
                    CheckboxListTile(
                      title: Text('기존 이미지 삭제', style: GoogleFonts.gaegu()),
                      value: _deleteExistingImage,
                      onChanged: (val) =>
                          setState(() => _deleteExistingImage = val ?? false),
                    ),
                  const SizedBox(height: 24),
                  _buildCareSection(),
                  const SizedBox(height: 24),
                  _buildNoteField(),
                ],
              ),
            ),
    );
  }

  // 이하 UI 빌드 위젯들은 그대로!
  Widget _buildDateChip(String formattedDate) {
    return Center(
      child: Text(
        formattedDate,
        style: GoogleFonts.gaegu(
          fontSize: 20,
          fontWeight: FontWeight.bold,
          color: Colors.grey[700],
        ),
      ),
    );
  }

  Widget _buildTitleField() {
    return TextFormField(
      controller: _titleController,
      decoration: InputDecoration(
        hintText: '✏️ 오늘의 일지 제목',
        hintStyle: GoogleFonts.gaegu(color: Colors.grey[500], fontSize: 24),
        border: InputBorder.none,
      ),
      style: GoogleFonts.gaegu(
        fontSize: 24,
        fontWeight: FontWeight.bold,
        color: const Color(0xFF3E2723),
      ),
    );
  }

  Widget _buildMyPlantSelection() {
    if (_myRegisteredPlants.isEmpty) {
      return Container(
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: Colors.grey[100],
          borderRadius: BorderRadius.circular(15),
        ),
        child: Center(
          child: Text(
            '등록된 식물이 없습니다.\n먼저 내 식물을 등록해주세요.',
            style: GoogleFonts.gaegu(),
            textAlign: TextAlign.center,
          ),
        ),
      );
    }
    return DropdownButtonFormField<String>(
      value: _selectedPlantNickname,
      hint: Text(
        '어떤 식물의 기록인가요?',
        style: GoogleFonts.gaegu(color: Colors.grey[600]),
      ),
      decoration: InputDecoration(
        prefixIcon: const Icon(
          Icons.filter_vintage_rounded,
          color: Color(0xFF2ECC71),
        ),
        filled: true,
        fillColor: Colors.green.withOpacity(0.05),
        border: OutlineInputBorder(
          borderRadius: BorderRadius.circular(15),
          borderSide: BorderSide.none,
        ),
      ),
      items: _myRegisteredPlants.map((plant) {
        return DropdownMenuItem(
          value: plant.plantNickname,
          child: Row(
            children: [
              ClipRRect(
                borderRadius: BorderRadius.circular(4),
                child: plant.userPlantImageUrl != null
                    ? Image.network(
                        plant.userPlantImageUrl!,
                        width: 24,
                        height: 24,
                        fit: BoxFit.cover,
                        errorBuilder: (c, e, s) => const Icon(
                          Icons.eco_rounded,
                          size: 24,
                          color: Colors.grey,
                        ),
                      )
                    : const Icon(
                        Icons.eco_rounded,
                        size: 24,
                        color: Colors.grey,
                      ),
              ),
              const SizedBox(width: 8),
              Text(plant.plantNickname ?? '이름 없음', style: GoogleFonts.gaegu()),
            ],
          ),
        );
      }).toList(),
      onChanged: (value) => setState(() => _selectedPlantNickname = value),
      style: GoogleFonts.gaegu(color: Colors.black87),
    );
  }

  Widget _buildImagePicker() {
    return GestureDetector(
      onTap: _pickImage,
      child: Container(
        height: 200,
        width: double.infinity,
        decoration: BoxDecoration(
          color: Colors.brown[50],
          borderRadius: BorderRadius.circular(15),
          border: Border.all(color: Colors.brown.withOpacity(0.2)),
          image: _imageFile != null
              ? DecorationImage(
                  image: FileImage(File(_imageFile!.path)),
                  fit: BoxFit.cover,
                )
              : null,
        ),
        child: _imageFile == null
            ? Center(
                child: Column(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Icon(
                      Icons.add_a_photo_outlined,
                      size: 40,
                      color: Colors.brown[300],
                    ),
                    const SizedBox(height: 8),
                    Text(
                      '사진 붙이기',
                      style: GoogleFonts.gaegu(color: Colors.brown[400]),
                    ),
                  ],
                ),
              )
            : null,
      ),
    );
  }

  Widget _buildCareSection() {
    return Row(
      mainAxisAlignment: MainAxisAlignment.spaceAround,
      children: [
        _buildCareItem(
          icon: Icons.water_drop_rounded,
          label: '물주기',
          isChecked: _watered,
          color: const Color(0xFF3498DB),
          onTap: () => setState(() => _watered = !_watered),
        ),
        _buildCareItem(
          icon: Icons.science_rounded,
          label: '영양제',
          isChecked: _fertilized,
          color: const Color(0xFFE67E22),
          onTap: () => setState(() => _fertilized = !_fertilized),
        ),
        _buildCareItem(
          icon: Icons.grass_rounded,
          label: '가지치기',
          isChecked: _pruned,
          color: const Color(0xFF9B59B6),
          onTap: () => setState(() => _pruned = !_pruned),
        ),
      ],
    );
  }

  Widget _buildCareItem({
    required IconData icon,
    required String label,
    required bool isChecked,
    required Color color,
    required VoidCallback onTap,
  }) {
    return GestureDetector(
      onTap: onTap,
      child: Column(
        children: [
          AnimatedContainer(
            duration: const Duration(milliseconds: 200),
            padding: const EdgeInsets.all(16),
            decoration: BoxDecoration(
              color: isChecked ? color : Colors.grey[200],
              shape: BoxShape.circle,
              boxShadow: isChecked
                  ? [
                      BoxShadow(
                        color: color.withOpacity(0.4),
                        blurRadius: 8,
                        offset: const Offset(0, 4),
                      ),
                    ]
                  : [],
            ),
            child: Icon(
              icon,
              size: 28,
              color: isChecked ? Colors.white : Colors.grey[600],
            ),
          ),
          const SizedBox(height: 8),
          Text(
            label,
            style: GoogleFonts.gaegu(
              fontWeight: FontWeight.bold,
              color: isChecked ? color : Colors.grey[600],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildNoteField() {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(15),
        border: Border.all(color: Colors.grey.shade200),
      ),
      child: TextFormField(
        controller: _notesController,
        maxLines: 10,
        decoration: InputDecoration(
          hintText:
              '오늘 식물은 어떤 모습이었나요?\n어떤 변화가 있었는지, 어떤 기분이 들었는지 자유롭게 기록해보세요...',
          hintStyle: GoogleFonts.gaegu(color: Colors.grey[400], height: 1.7),
          border: InputBorder.none,
        ),
        style: GoogleFonts.gaegu(
          fontSize: 16,
          height: 1.7,
          color: const Color(0xFF3E2723),
        ),
      ),
    );
  }
}

File: apps/mobile/lib/screens/NoteScreen.java
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:image_picker/image_picker.dart';
import 'package:intl/intl.dart';
import 'dart:io';
import '../models/note_model.dart';

class NoteScreen extends StatefulWidget {
  final DateTime selectedDate;
  const NoteScreen({super.key, required this.selectedDate});

  @override
  State<NoteScreen> createState() => _NoteScreenState();
}

class _NoteScreenState extends State<NoteScreen> with TickerProviderStateMixin {
  XFile? _imageFile;
  bool _watered = false;
  bool _fertilized = false;
  bool _pruned = false; // 가지치기 상태 변수 추가
  final TextEditingController _notesController = TextEditingController();
  final ImagePicker _picker = ImagePicker();
  late AnimationController _animationController;
  late Animation<double> _slideAnimation;

  @override
  void initState() {
    super.initState();
    _animationController = AnimationController(
      duration: const Duration(milliseconds: 500),
      vsync: this,
    );
    _slideAnimation = Tween<double>(begin: 0.0, end: 1.0).animate(
      CurvedAnimation(parent: _animationController, curve: Curves.easeOut),
    );
    _animationController.forward();
  }

  @override
  void dispose() {
    _animationController.dispose();
    _notesController.dispose();
    super.dispose();
  }

  Future<void> _pickImage() async {
    showModalBottomSheet(
      context: context,
      backgroundColor: Colors.transparent,
      builder: (context) => Container(
        padding: const EdgeInsets.all(20),
        decoration: const BoxDecoration(
          color: Colors.white,
          borderRadius: BorderRadius.vertical(top: Radius.circular(20)),
        ),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Container(
              width: 40,
              height: 4,
              decoration: BoxDecoration(
                color: Colors.grey[300],
                borderRadius: BorderRadius.circular(2),
              ),
            ),
            const SizedBox(height: 20),
            Text(
              '사진 선택',
              style: GoogleFonts.gaegu(
                fontSize: 20,
                fontWeight: FontWeight.bold,
              ),
            ),
            const SizedBox(height: 20),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
              children: [
                _buildImageSourceOption(
                  icon: Icons.camera_alt,
                  label: '카메라',
                  onTap: () async {
                    Navigator.pop(context);
                    final pickedFile = await _picker.pickImage(
                      source: ImageSource.camera,
                    );
                    if (pickedFile != null) {
                      setState(() {
                        _imageFile = pickedFile;
                      });
                    }
                  },
                ),
                _buildImageSourceOption(
                  icon: Icons.photo_library,
                  label: '갤러리',
                  onTap: () async {
                    Navigator.pop(context);
                    final pickedFile = await _picker.pickImage(
                      source: ImageSource.gallery,
                    );
                    if (pickedFile != null) {
                      setState(() {
                        _imageFile = pickedFile;
                      });
                    }
                  },
                ),
              ],
            ),
            const SizedBox(height: 20),
          ],
        ),
      ),
    );
  }

  Widget _buildImageSourceOption({
    required IconData icon,
    required String label,
    required VoidCallback onTap,
  }) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        padding: const EdgeInsets.all(20),
        decoration: BoxDecoration(
          color: const Color(0xFF2ECC71).withOpacity(0.1),
          borderRadius: BorderRadius.circular(16),
        ),
        child: Column(
          children: [
            Icon(icon, size: 32, color: const Color(0xFF2ECC71)),
            const SizedBox(height: 8),
            Text(
              label,
              style: GoogleFonts.gaegu(
                color: const Color(0xFF2ECC71),
                fontWeight: FontWeight.bold,
              ),
            ),
          ],
        ),
      ),
    );
  }

  void _saveNote() {
    if (_notesController.text.trim().isEmpty && _imageFile == null) {
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(
          content: Text(
            '사진이나 글 중 하나는 꼭 기록해주세요!',
            style: GoogleFonts.gaegu(),
          ),
          backgroundColor: Colors.red[400],
          behavior: SnackBarBehavior.floating,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(10),
          ),
        ),
      );
      return;
    }

    final newNote = Note(
      date: widget.selectedDate,
      image: _imageFile,
      watered: _watered,
      fertilized: _fertilized,
      pruned: _pruned, // 저장할 때 가지치기 정보 포함
      text: _notesController.text.trim(),
    );

    Navigator.pop(context, newNote);
  }

  @override
  Widget build(BuildContext context) {
    String formattedDate = DateFormat('yyyy. MM. dd (E)', 'ko_KR').format(widget.selectedDate);

    return Scaffold(
      backgroundColor: Colors.grey[50],
      appBar: AppBar(
        title: Text(
          '성장 일지 작성',
          style: GoogleFonts.gaegu(fontWeight: FontWeight.bold, fontSize: 24),
        ),
        backgroundColor: const Color(0xFF2ECC71),
        foregroundColor: Colors.white,
        elevation: 0,
        flexibleSpace: Container(
          decoration: const BoxDecoration(
            gradient: LinearGradient(
              colors: [Color(0xFF2ECC71), Color(0xFF27AE60)],
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
            ),
          ),
        ),
        actions: [
          Container(
            margin: const EdgeInsets.only(right: 12, top: 8, bottom: 8),
            child: ElevatedButton(
              onPressed: _saveNote,
              style: ElevatedButton.styleFrom(
                backgroundColor: Colors.white,
                foregroundColor: const Color(0xFF2ECC71),
                elevation: 0,
                shape: RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(20),
                ),
              ),
              child: Text(
                '저장',
                style: GoogleFonts.gaegu(
                  fontSize: 16,
                  fontWeight: FontWeight.bold,
                ),
              ),
            ),
          ),
        ],
      ),
      body: SlideTransition(
        position: Tween<Offset>(
          begin: const Offset(0, 0.3),
          end: Offset.zero,
        ).animate(_slideAnimation),
        child: FadeTransition(
          opacity: _slideAnimation,
          child: SingleChildScrollView(
            padding: const EdgeInsets.all(20),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                _buildDateChip(formattedDate),
                const SizedBox(height: 24),
                _buildImageSection(),
                const SizedBox(height: 24),
                _buildCareSection(),
                const SizedBox(height: 24),
                _buildNoteSection(),
                const SizedBox(height: 100),
              ],
            ),
          ),
        ),
      ),
    );
  }

  Widget _buildDateChip(String formattedDate) {
    return Center(
      child: Container(
        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 12),
        decoration: BoxDecoration(
          gradient: const LinearGradient(
            colors: [Color(0xFF2ECC71), Color(0xFF27AE60)],
          ),
          borderRadius: BorderRadius.circular(25),
          boxShadow: [
            BoxShadow(
              color: const Color(0xFF2ECC71).withOpacity(0.3),
              blurRadius: 10,
              offset: const Offset(0, 5),
            ),
          ],
        ),
        child: Text(
          formattedDate,
          style: GoogleFonts.gaegu(
            fontSize: 18,
            fontWeight: FontWeight.bold,
            color: Colors.white,
          ),
        ),
      ),
    );
  }

  Widget _buildImageSection() {
    return Container(
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(20),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withOpacity(0.1),
            blurRadius: 20,
            offset: const Offset(0, 10),
          ),
        ],
      ),
      child: ClipRRect(
        borderRadius: BorderRadius.circular(20),
        child: GestureDetector(
          onTap: _pickImage,
          child: Container(
            height: 280,
            width: double.infinity,
            decoration: BoxDecoration(
              color: _imageFile != null ? null : Colors.grey[100],
              image: _imageFile != null
                  ? DecorationImage(
                image: FileImage(File(_imageFile!.path)),
                fit: BoxFit.cover,
              )
                  : null,
            ),
            child: _imageFile == null
                ? Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Container(
                  padding: const EdgeInsets.all(20),
                  decoration: BoxDecoration(
                    color: const Color(0xFF2ECC71).withOpacity(0.1),
                    shape: BoxShape.circle,
                  ),
                  child: const Icon(
                    Icons.add_a_photo_outlined,
                    size: 48,
                    color: Color(0xFF2ECC71),
                  ),
                ),
                const SizedBox(height: 16),
                Text(
                  '사진 추가하기',
                  style: GoogleFonts.gaegu(
                    fontSize: 18,
                    fontWeight: FontWeight.bold,
                    color: const Color(0xFF2C3E50),
                  ),
                ),
                const SizedBox(height: 8),
                Text(
                  '식물의 성장 모습을 기록해보세요',
                  style: GoogleFonts.gaegu(
                    fontSize: 14,
                    color: const Color(0xFF7F8C8D),
                  ),
                ),
              ],
            )
                : Stack(
              children: [
                Positioned(
                  top: 12,
                  right: 12,
                  child: GestureDetector(
                    onTap: () {
                      setState(() {
                        _imageFile = null;
                      });
                    },
                    child: Container(
                      padding: const EdgeInsets.all(8),
                      decoration: const BoxDecoration(
                        color: Colors.black54,
                        shape: BoxShape.circle,
                      ),
                      child: const Icon(
                        Icons.close,
                        color: Colors.white,
                        size: 20,
                      ),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }

  Widget _buildCareSection() {
    return Container(
      padding: const EdgeInsets.all(20),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(20),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withOpacity(0.1),
            blurRadius: 20,
            offset: const Offset(0, 10),
          ),
        ],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              Container(
                padding: const EdgeInsets.all(8),
                decoration: BoxDecoration(
                  color: const Color(0xFF2ECC71).withOpacity(0.1),
                  borderRadius: BorderRadius.circular(10),
                ),
                child: const Icon(
                  Icons.eco,
                  color: Color(0xFF2ECC71),
                  size: 24,
                ),
              ),
              const SizedBox(width: 12),
              Text(
                '식물 관리',
                style: GoogleFonts.gaegu(
                  fontSize: 20,
                  fontWeight: FontWeight.bold,
                  color: const Color(0xFF2C3E50),
                ),
              ),
            ],
          ),
          const SizedBox(height: 20),
          Row(
            children: [
              Expanded(
                child: _buildCareItem(
                  icon: Icons.water_drop,
                  label: '물 주기',
                  isChecked: _watered,
                  color: const Color(0xFF3498DB),
                  onTap: () {
                    setState(() {
                      _watered = !_watered;
                    });
                  },
                ),
              ),
              const SizedBox(width: 16),
              Expanded(
                child: _buildCareItem(
                  icon: Icons.science,
                  label: '영양제',
                  isChecked: _fertilized,
                  color: const Color(0xFFE67E22),
                  onTap: () {
                    setState(() {
                      _fertilized = !_fertilized;
                    });
                  },
                ),
              ),
              const SizedBox(width: 16),
              // 👇 가지치기 아이템 추가
              Expanded(
                child: _buildCareItem(
                  icon: Icons.content_cut,
                  label: '가지치기',
                  isChecked: _pruned,
                  color: const Color(0xFF9B59B6),
                  onTap: () {
                    setState(() {
                      _pruned = !_pruned;
                    });
                  },
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildCareItem({
    required IconData icon,
    required String label,
    required bool isChecked,
    required Color color,
    required VoidCallback onTap,
  }) {
    return GestureDetector(
      onTap: onTap,
      child: AnimatedContainer(
        duration: const Duration(milliseconds: 200),
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: isChecked ? color.withOpacity(0.1) : Colors.grey[50],
          borderRadius: BorderRadius.circular(16),
          border: Border.all(
            color: isChecked ? color : Colors.grey[300]!,
            width: 2,
          ),
        ),
        child: Column(
          children: [
            AnimatedContainer(
              duration: const Duration(milliseconds: 200),
              padding: const EdgeInsets.all(12),
              decoration: BoxDecoration(
                color: isChecked ? color : Colors.grey[300],
                shape: BoxShape.circle,
              ),
              child: Icon(
                icon,
                size: 24,
                color: isChecked ? Colors.white : Colors.grey[600],
              ),
            ),
            const SizedBox(height: 8),
            Text(
              label,
              style: GoogleFonts.gaegu(
                fontSize: 16,
                fontWeight: FontWeight.bold,
                color: isChecked ? color : Colors.grey[600],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildNoteSection() {
    return Container(
      padding: const EdgeInsets.all(20),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(20),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withOpacity(0.1),
            blurRadius: 20,
            offset: const Offset(0, 10),
          ),
        ],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              Container(
                padding: const EdgeInsets.all(8),
                decoration: BoxDecoration(
                  color: const Color(0xFF2ECC71).withOpacity(0.1),
                  borderRadius: BorderRadius.circular(10),
                ),
                child: const Icon(
                  Icons.edit_note,
                  color: Color(0xFF2ECC71),
                  size: 24,
                ),
              ),
              const SizedBox(width: 12),
              Text(
                '오늘의 기록',
                style: GoogleFonts.gaegu(
                  fontSize: 20,
                  fontWeight: FontWeight.bold,
                  color: const Color(0xFF2C3E50),
                ),
              ),
            ],
          ),
          const SizedBox(height: 16),
          Container(
            decoration: BoxDecoration(
              color: Colors.grey[50],
              borderRadius: BorderRadius.circular(16),
              border: Border.all(color: Colors.grey[300]!),
            ),
            child: TextFormField(
              controller: _notesController,
              maxLines: 8,
              decoration: InputDecoration(
                hintText: '오늘 식물은 어떤가요?\n\n• 새로운 변화나 성장\n• 잎의 색깔이나 모양\n• 특별한 관찰 내용\n• 느낀 점이나 기분',
                hintStyle: GoogleFonts.gaegu(
                  color: Colors.grey[500],
                  fontSize: 16,
                  height: 1.5,
                ),
                border: InputBorder.none,
                contentPadding: const EdgeInsets.all(16),
              ),
              style: GoogleFonts.gaegu(
                fontSize: 16,
                height: 1.6,
                color: const Color(0xFF2C3E50),
              ),
            ),
          ),
          const SizedBox(height: 12),
          Row(
            children: [
              Icon(
                Icons.lightbulb_outline,
                size: 16,
                color: Colors.grey[600],
              ),
              const SizedBox(width: 8),
              Expanded(
                child: Text(
                  '작은 변화도 소중한 성장의 기록이에요',
                  style: GoogleFonts.gaegu(
                    fontSize: 14,
                    color: Colors.grey[600],
                  ),
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }
}
